### PR TITLE
Bump kubernetes dependency to 1.17 and golang to 1.13

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test

 /kind feature

**What this PR does / why we need it**:
Upgrade the k8s dependency  to 1.17 and golang to 1.13.
Will update the Compatibility matrix in follow up PR.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Bump kubernetes dependency to 1.17 and golang to 1.13
```

### CODE REVIEW

**Comment by fisherxu** [comment]
Ready for review now :)  cc @kevin-wangzefeng @sids-b @kadisi @chendave @subpathdev  PTAL, thanks!

**Comment by kadisi** [comment]
after this change , Can we only compile with go 1.13？

**Comment by fisherxu** [comment]
> after this change , Can we only compile with go 1.13？

Yes, if we use go1.12 and enable gomod, there will be some checksum error. But using vendor to compile is OK.

And using go1.13, it's ok to use both vendor and go mod.

**Comment by kadisi** [comment]
> Yes, if we use go1.12 and enable gomod, there will be some checksum error. But using vendor to compile is OK.

@fisherxu could you please show those checksum error ??

**Comment by fisherxu** [comment]
> @fisherxu could you please show those checksum error ??

Here is the error log using go1.12
```
➜  kubeedge git:(upgrade-17) ✗ make
cd cloud && make
make[1]: Entering directory '/root/codes/src/github.com/kubeedge/kubeedge/cloud'
bash ../hack/verify-golang.sh
go detail version: go version go1.12 linux/amd64
go version: 1.12
go build -ldflags '-X github.com/kubeedge/kubeedge/pkg/version.buildDate=2020-01-03T02:31:57Z -X github.com/kubeedge/kubeedge
/pkg/version.gitCommit=12544d8e6aa3498fd5d9c8ea0e19661c9dc312ac -X github.com/kubeedge/kubeedge/pkg/version.gitTreeState=dirt
y -X github.com/kubeedge/kubeedge/pkg/version.gitVersion=v1.1.0-beta.0.318+12544d8e6aa349-dirty -X github.com/kubeedge/kubeed
ge/pkg/version.gitMajor=1 -X github.com/kubeedge/kubeedge/pkg/version.gitMinor=1+' cmd/cloudcore/cloudcore.go
go: finding github.com/mailru/easyjson v0.7.0
go: finding github.com/prometheus/procfs v0.0.3
go: finding github.com/mailru/easyjson v0.0.0-20160728113105-d5b7844b561a
go: finding github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1
go: finding github.com/jtolds/gls v4.20.0+incompatible
go: finding github.com/go-logr/logr v0.1.0
verifying github.com/apache/servicecomb-kie@v0.1.0/go.mod: checksum mismatch
downloaded: h1:bTPGCj54qKIZtNTDsP6FcBZqp5BsscfSDDwl5SKmZfo=
go.sum:     h1:UbinHy4mdkrTqM71CmmDtc/xGyq46mud04OgprdETkM=
Makefile:11: recipe for target 'cloudcore' failed
make[1]: *** [cloudcore] Error 1
make[1]: Leaving directory '/root/codes/src/github.com/kubeedge/kubeedge/cloud'
Makefile:5: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by kadisi** [comment]
this problem maybe related to `go-chassis/ go-chassis` package.

**Comment by fisherxu** [comment]
With this fix, now it can compile with go 1.12 and 1.13, and I also have updated the Compatibility matrix,  PTAL :)

File: cloud/pkg/devicecontroller/utils/crdclient.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func NewCRDClient(cfg *rest.Config) (*rest.RESTClient, error) {
 	config.APIPath = "/apis"
 	config.GroupVersion = &v1alpha1.SchemeGroupVersion
 	config.ContentType = runtime.ContentTypeJSON
-	config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: serializer.NewCodecFactory(scheme)}
+	config.NegotiatedSerializer = serializer.NewCodecFactory(scheme).WithoutConversion()
```
Comment (chendave) [inline cloud/pkg/devicecontroller/utils/crdclient.go]: change needed from k8s 1.17? what's will happen without this change?  could you pls say something in the commit message? thanks!

File: edge/pkg/edged/edged.go [hunk 2]
```diff
@@ -50,9 +50,10 @@ import (
 	"k8s.io/client-go/util/workqueue"
 	internalapi "k8s.io/cri-api/pkg/apis"
 	"k8s.io/klog"
+	pluginwatcherapi "k8s.io/kubelet/pkg/apis/pluginregistration/v1"
```
Comment (chendave) [inline edge/pkg/edged/edged.go]: I don't know to comment on the entire PR or files, I assume all those changes are needed due to k8s version upgrade, right?

Comment (fisherxu) [reply edge/pkg/edged/edged.go]: Yes, we import the k8s api, so must adapt to its changes :)

File: cloud/pkg/devicecontroller/utils/crdclient.go (hunk 1 above)
Comment (fisherxu) [reply cloud/pkg/devicecontroller/utils/crdclient.go]: Yes, there's some refactor in k8s.io/apimachinery, without this change, the crd client can't talk to kube-apiserver.

**Comment by fisherxu** [comment]
>  could you pls say something in the commit message? 

Have done.

**Comment by subpathdev** [comment]
pls fix typo in the commit message: `upodate the Compatibility matrix`
/lgtm cancel

**Comment by fisherxu** [comment]
> pls fix typo in the commit message: upodate the Compatibility matrix

Done. @subpathdev

**Pull request merged**